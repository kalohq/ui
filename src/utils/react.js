import ReactDOM from 'react-dom';

/**
 * Focus a ref’d react input when it is mounted
 * @param ref
 */
export function focusOnMount(ref) {
  const el = ReactDOM.findDOMNode(ref);

  if (el) {
    // Give focus and put the cursor at the end of
    // the value.
    el.focus();
    el.value = el.value;
  }
}

/**
 * Select the text of a ref’d react input when it is mounted
 * @param ref
 */
export function selectOnMount(ref) {
  if (ref && ref.select) {
    ref.select();
  }
}
