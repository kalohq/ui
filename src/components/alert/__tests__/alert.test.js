/* eslint-env jest */
import * as React from 'react';
import renderer from 'react-test-renderer';

import Alert from 'components/alert';

describe('Alert', () => {
  const defaultProps = {
    type: 'info',
  };
  const create = (props = {}) =>
    renderer
      .create(
        <Alert
          {...defaultProps}
          {...props}
          children={`A ${props.type || defaultProps.type} alert`}
        />
      )
      .toJSON();

  test('should render shallow component ok', () => {
    const element = create();
    expect(element).toMatchSnapshot();
  });

  test('should render a warning alert', () => {
    const element = create({
      type: 'warning',
    });
    expect(element).toMatchSnapshot();
  });

  test('should render a confirmation alert with an icon', () => {
    const element = create({
      type: 'confirmation',
      showIcon: true,
    });
    expect(element).toMatchSnapshot();
  });
});
