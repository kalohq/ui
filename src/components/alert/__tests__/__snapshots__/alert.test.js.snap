// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Alert should render a confirmation alert with an icon 1`] = `
.emotion-5 {
  position: relative;
  -webkit-flex-direction: row;
  -ms-flex-direction: row;
  flex-direction: row;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-flex-shrink: 0;
  -ms-flex-negative: 0;
  flex-shrink: 0;
  -webkit-align-content: flex-start;
  -ms-flex-line-pack: flex-start;
  align-content: flex-start;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  padding: 8px 16px;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  border-radius: 4px;
  background-color: rgb(229,247,241);
}

.emotion-1 {
  display: inline-block;
  vertical-align: bottom;
  margin-right: 8px;
  -webkit-user-select: none;
  -moz-user-select: none;
  -ms-user-select: none;
  user-select: none;
  line-height: 1em;
  vertical-align: middle;
  cursor: inherit;
  font-size: 16px;
  height: 16px;
  fill: rgb(62,179,138);
}

.emotion-3 {
  display: inline-block;
  vertical-align: bottom;
  font-family: 'WebFaktSoftPro',sans-serif;
  white-space: nowrap;
  text-overflow: ellipsis;
  overflow: hidden;
  cursor: inherit;
  text-align: unset;
  text-decoration: inherit;
  font-weight: 400;
  font-size: 14px;
  color: rgb(62,179,138);
  white-space: normal;
  text-overflow: initial;
  overflow: visible;
  white-space: pre-wrap;
  word-break: break-word;
}

.emotion-3:hover {
  text-decoration: inherit;
}

.emotion-3 strong {
  font-weight: 500;
}

<div
  className="emotion-4 emotion-5"
  role="alert"
  style={Object {}}
  type="confirmation"
>
  <i
    className="emotion-0 emotion-1"
    color="green"
    onClick={undefined}
    size={16}
    style={Object {}}
    title={undefined}
  >
    <svg
      aria-hidden="true"
      height={16}
      width={16}
    >
      <title>
        check_circle
      </title>
      <use
        xlinkHref="#check_circle"
      />
    </svg>
  </i>
  <span
    className="emotion-2 emotion-3"
    color="green"
    onClick={undefined}
    size="small"
    style={Object {}}
  >
    A confirmation alert
  </span>
</div>
`;

exports[`Alert should render a warning alert 1`] = `
.emotion-3 {
  position: relative;
  -webkit-flex-direction: row;
  -ms-flex-direction: row;
  flex-direction: row;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-flex-shrink: 0;
  -ms-flex-negative: 0;
  flex-shrink: 0;
  -webkit-align-content: flex-start;
  -ms-flex-line-pack: flex-start;
  align-content: flex-start;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  padding: 8px 16px;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  border-radius: 4px;
  background-color: rgb(255,246,216);
}

.emotion-1 {
  display: inline-block;
  vertical-align: bottom;
  font-family: 'WebFaktSoftPro',sans-serif;
  white-space: nowrap;
  text-overflow: ellipsis;
  overflow: hidden;
  cursor: inherit;
  text-align: unset;
  text-decoration: inherit;
  font-weight: 400;
  font-size: 14px;
  color: rgb(253,184,28);
  white-space: normal;
  text-overflow: initial;
  overflow: visible;
  white-space: pre-wrap;
  word-break: break-word;
}

.emotion-1:hover {
  text-decoration: inherit;
}

.emotion-1 strong {
  font-weight: 500;
}

<div
  className="emotion-2 emotion-3"
  role="alert"
  style={Object {}}
  type="warning"
>
  <span
    className="emotion-0 emotion-1"
    color="orange"
    onClick={undefined}
    size="small"
    style={Object {}}
  >
    A warning alert
  </span>
</div>
`;

exports[`Alert should render shallow component ok 1`] = `
.emotion-3 {
  position: relative;
  -webkit-flex-direction: row;
  -ms-flex-direction: row;
  flex-direction: row;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-flex-shrink: 0;
  -ms-flex-negative: 0;
  flex-shrink: 0;
  -webkit-align-content: flex-start;
  -ms-flex-line-pack: flex-start;
  align-content: flex-start;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  padding: 8px 16px;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  border-radius: 4px;
  background-color: rgb(232,250,252);
}

.emotion-1 {
  display: inline-block;
  vertical-align: bottom;
  font-family: 'WebFaktSoftPro',sans-serif;
  white-space: nowrap;
  text-overflow: ellipsis;
  overflow: hidden;
  cursor: inherit;
  text-align: unset;
  text-decoration: inherit;
  font-weight: 400;
  font-size: 14px;
  color: rgb(84,195,219);
  white-space: normal;
  text-overflow: initial;
  overflow: visible;
  white-space: pre-wrap;
  word-break: break-word;
}

.emotion-1:hover {
  text-decoration: inherit;
}

.emotion-1 strong {
  font-weight: 500;
}

<div
  className="emotion-2 emotion-3"
  role="alert"
  style={Object {}}
  type="info"
>
  <span
    className="emotion-0 emotion-1"
    color="blue"
    onClick={undefined}
    size="small"
    style={Object {}}
  >
    A info alert
  </span>
</div>
`;
